solr/core/src/java/org/apache/solr/pkg/PackageLoader.java

```diff
@@ -301,6 +295,32 @@ public String toString() {
       }
     }
   }
+  static class PackageResourceLoader extends SolrResourceLoader {
+
+    PackageResourceLoader(String name, List<Path> classpath, Path instanceDir, ClassLoader parent) {
+      super(name, classpath, instanceDir, parent);
+    }
+
+    @Override
+    public <T> boolean addToCoreAware(T obj) {
+      //do not do anything
+      //this class is
+      return false;
+    }
+
+    @Override
+    public <T> boolean addToResourceLoaderAware(T obj) {
+      if (obj instanceof ResourceLoaderAware) {
+        assertAwareCompatibility(ResourceLoaderAware.class, obj);
+        try {
+          ((ResourceLoaderAware) obj).inform(this);
```

Comment:
Lets figure out why the base class doesn't just immediately call inform like this here.  What test then fails?  I've read an interesting JIRA issue: https://issues.apache.org/jira/browse/SOLR-8311  (lots of info there; I recommend reading all of it)  It's been on my mind for some time now actually since when I was cleaning up SolrResourceLoader since this "live" wart seems like a hack.